controller/src/main/java/io/pravega/controller/store/stream/StoreException.java

```diff
@@ -177,6 +177,15 @@ private IllegalStateException(String errorMessage, Throwable cause) {
         }
     }
 
+    /**
+     * Exception type when you are attempting an illegal operation which should not be retried.
+     */
+    public static class IllegalStreamStateException extends StoreException {
```

Comment:
With the changed code I don't see the need to create this new exception. You can directly use StoreException.IllegalStateException() and provide the appropriate message.
That is a non-retryable exception. 
So the event should not get retried.